qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -595,15 +595,15 @@ class ScalableSymbolicPulse(SymbolicPulse):
         additional_msg=(
             "Instead, use a float for ``amp`` (for the magnitude) and a float for ``angle``"
         ),
-        since="0.23.0",
-        pending=True,
+        since="0.25.0",
+        pending=False,
         predicate=lambda amp: isinstance(amp, complex),
     )
     def __init__(
         self,
         pulse_type: str,
         duration: Union[ParameterExpression, int],
-        amp: Union[ParameterExpression, float, complex],
+        amp: Union[ParameterExpression, float],
```

Comment:
```suggestion
        amp: ParameterValueType,
```
just a nitpick. Do you also want to replace the type hint  of `angle` and `parameters`?

Reply:
Nice catch. `angle` can change as well. But `parameters` is maybe a bit more tricky, because we didn't deprecate complex values there. Only in `amp`. What do you think? Should we add another deprecation warning for `parameters`? Should we do it here or in a separate PR?

Reply:
Thanks @nkanazawa1989.
I'll change the hints where we can, and resolve the conflicts created by #10222.